backend/libinput/pointer.c

```diff
@@ -92,43 +137,32 @@ void handle_pointer_axis(struct libinput_event *event,
 		wlr_log(L_DEBUG, "Got a pointer event for a device with no pointers?");
 		return;
 	}
+
 	struct libinput_event_pointer *pevent =
 		libinput_event_get_pointer_event(event);
-	struct wlr_event_pointer_axis wlr_event = { 0 };
-	wlr_event.device = wlr_dev;
-	wlr_event.time_msec =
-		usec_to_msec(libinput_event_pointer_get_time_usec(pevent));
-	switch (libinput_event_pointer_get_axis_source(pevent)) {
-	case LIBINPUT_POINTER_AXIS_SOURCE_WHEEL:
-		wlr_event.source = WLR_AXIS_SOURCE_WHEEL;
-		break;
-	case LIBINPUT_POINTER_AXIS_SOURCE_FINGER:
-		wlr_event.source = WLR_AXIS_SOURCE_FINGER;
-		break;
-	case LIBINPUT_POINTER_AXIS_SOURCE_CONTINUOUS:
-		wlr_event.source = WLR_AXIS_SOURCE_CONTINUOUS;
-		break;
-	case LIBINPUT_POINTER_AXIS_SOURCE_WHEEL_TILT:
-		wlr_event.source = WLR_AXIS_SOURCE_WHEEL_TILT;
-		break;
-	}
+	enum libinput_pointer_axis_source source =
+		libinput_event_pointer_get_axis_source(pevent);
+	uint64_t time_usec = libinput_event_pointer_get_time_usec(pevent);
+
+	struct wlr_event_pointer_axis wlr_event = {
+		.device = wlr_dev,
+		.time_msec = usec_to_msec(time_usec),
+		.source = axis_source_to_wlr(source),
+	};
+
 	enum libinput_pointer_axis axies[] = {
 		LIBINPUT_POINTER_AXIS_SCROLL_VERTICAL,
 		LIBINPUT_POINTER_AXIS_SCROLL_HORIZONTAL,
 	};
+
 	for (size_t i = 0; i < sizeof(axies) / sizeof(axies[0]); ++i) {
```

Comment:
The plural of `axis` is `axes`. Or is there another version of the plural form I'm not aware of?

Reply:
You're correct. This is incorrect.